packages/get-credentials/test/get-credentials.spec.js

```diff
@@ -0,0 +1,104 @@
+import fs from 'fs'
+import sinon from 'sinon'
+
+import {
+  getCredentials,
+  getCredentialsFromEnvironment,
+  setCredentialsFromEnvFile,
+} from '../src/index'
+import { homepage } from '../package.json'
+
+describe('getCredentials', () => {
+  test('should reject error when missing project key argument', (done) => {
+    getCredentials()
+      .then(done.fail)
+      .catch((errors) => {
+        expect(errors[0].message).toBe('Project Key is needed')
+        done()
+      })
+  })
+
+  test('should reject an array of all errors', (done) => {
+    getCredentials('forgery')
+      .then(done.fail)
+      .catch((errors) => {
+        expect(errors.length).toBe(2)
+        expect(errors[0].message).toBeTruthy()
+        expect(errors[1].message).toBeTruthy()
+        done()
+      })
+  })
+})
+
+describe('getCredentialsFromEnvironment', () => {
+  const sandbox = sinon.sandbox.create()
+
+  afterEach(() => sandbox.restore())
+
+  test('should resolve credentials from environment variables', () => {
+    sandbox.stub(process, 'env', { CT_STROOPWAFEL: 'nyw:les' })
```

Comment:
The reason I went with a sandbox is that it doesn't influence the other tests. Doing it without a sandbox is a bit dangerous because one test can influence the result of others. Also relying on `process.env` without a sandbox means that developer's environment variables could collide with the tests.